Tuist/ProjectDescriptionHelpers/Module.swift

```diff
@@ -0,0 +1,643 @@
+
+        return targets + testTargets
+    }
+
+    public var sourceTargets: [Target] {
+        let isStaticProduct = product == .staticLibrary || product == .staticFramework
+
+        return [
+            .target(
+                name: targetName,
+                product: product,
+                dependencies: dependencies + (isStaticProduct ? [
+                    .external(name: "Mockable"),
+                ] : [])
+            ),
+        ]
+    }
+
+    fileprivate var sharedDependencies: [TargetDependency] {
+        return [
+            .external(name: "SwiftToolsSupport"),
+            .external(name: "SystemPackage"),
+        ]
+    }
+
+    public var acceptanceTestsTargetName: String? {
+        switch self {
+        case .tuist, .automation, .dependencies, .generator:
+            return "\(rawValue)AcceptanceTests"
+        default:
+            return nil
+        }
+    }
+
+    public var testingTargetName: String? {
+        switch self {
+        case .tuist, .tuistBenchmark, .tuistFixtureGenerator, .kit, .projectAutomation, .projectDescription, .analytics,
+             .dependencies, .acceptanceTesting:
+            return nil
+        default:
+            return "\(rawValue)Testing"
+        }
+    }
+
+    public var unitTestsTargetName: String? {
+        switch self {
+        case .automation, .analytics, .tuist, .tuistBenchmark, .tuistFixtureGenerator, .projectAutomation, .projectDescription,
+             .acceptanceTesting:
+            return nil
+        default:
+            return "\(rawValue)Tests"
+        }
+    }
+
+    public var integrationTestsTargetName: String? {
+        switch self {
+        case .tuist, .tuistBenchmark, .tuistFixtureGenerator, .projectAutomation, .projectDescription, .graph, .asyncQueue,
+             .plugin, .analytics, .dependencies, .acceptanceTesting:
+            return nil
+        default:
+            return "\(rawValue)IntegrationTests"
+        }
+    }
+
+    public var targetName: String {
+        rawValue
+    }
+
+    public var product: Product {
+        switch self {
+        case .tuist, .tuistBenchmark, .tuistFixtureGenerator:
+            return .commandLineTool
+        case .projectAutomation, .projectDescription:
+            return .framework
+        default:
+            return .staticFramework
+        }
+    }
+
+    public var acceptanceTestDependencies: [TargetDependency] {
+        let dependencies: [TargetDependency] = switch self {
+        case .tuist, .automation, .dependencies, .generator:
+            [
+                .target(name: "TuistAcceptanceTesting"),
+                .target(name: "TuistSupportTesting"),
+            ]
+        default:
+            []
+        }
+        return dependencies + sharedDependencies
+    }
+
+    public var dependencies: [TargetDependency] {
+        var dependencies: [TargetDependency] = switch self {
+        case .acceptanceTesting:
+            [
+                .target(name: "TuistKit"),
```

Comment:
Should we use 
```suggestion
                .target(name: Module.kit.rawString),
```
?

Reply:
I pushed a commit, removing all the strings from dependencies. It's type-safer now 🎉